File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/UpdateJobDetails.java
package com.oracle.aidataplatform.dp.model;



/**
 * The data to update a job.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UpdateJobDetails.Builder.class)

public final class UpdateJobDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"runAs", "name", "description", "schedule", "continuous", "maxConcurrentRuns", "gitConfig", "queue", "jobClusters", "path", "tasks", "parameters", "timeoutSeconds"})
    public UpdateJobDetails(String runAs, String name, String description, Schedule schedule, Continuous continuous, Integer maxConcurrentRuns, GitConfig gitConfig, Queue queue, java.util.List<JobCluster> jobClusters, String path, java.util.List<Task> tasks, java.util.List<Parameter> parameters, Integer timeoutSeconds) {
        super();
        this.runAs = runAs;
        this.name = name;
        this.description = description;
        this.schedule = schedule;
        this.continuous = continuous;
        this.maxConcurrentRuns = maxConcurrentRuns;
        this.gitConfig = gitConfig;
        this.queue = queue;
        this.jobClusters = jobClusters;
        this.path = path;
        this.tasks = tasks;
        this.parameters = parameters;
        this.timeoutSeconds = timeoutSeconds;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The id with which the job run as.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("runAs")
private String runAs;

        /**
         * The id with which the job run as.
         * @param runAs the value to set
         * @return this builder
         **/
        

public Builder runAs(String runAs) {
    this.runAs = runAs;
    return this;
}
            /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * A user-friendly name. Does not have to be unique, and is changeable.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * A description for the job.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * A description for the job.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("schedule")
private Schedule schedule;



public Builder schedule(Schedule schedule) {
    this.schedule = schedule;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("continuous")
private Continuous continuous;



public Builder continuous(Continuous continuous) {
    this.continuous = continuous;
    return this;
}
            /**
     * Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("maxConcurrentRuns")
private Integer maxConcurrentRuns;

        /**
         * Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.
         * @param maxConcurrentRuns the value to set
         * @return this builder
         **/
        

public Builder maxConcurrentRuns(Integer maxConcurrentRuns) {
    this.maxConcurrentRuns = maxConcurrentRuns;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("gitConfig")
private GitConfig gitConfig;



public Builder gitConfig(GitConfig gitConfig) {
    this.gitConfig = gitConfig;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("queue")
private Queue queue;



public Builder queue(Queue queue) {
    this.queue = queue;
    return this;
}
            /**
     * List of job cluster configurations.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("jobClusters")
private java.util.List<JobCluster> jobClusters;

        /**
         * List of job cluster configurations.
         * @param jobClusters the value to set
         * @return this builder
         **/
        

public Builder jobClusters(java.util.List<JobCluster> jobClusters) {
    this.jobClusters = jobClusters;
    return this;
}
            /**
     * The path to store the job definition in.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("path")
private String path;

        /**
         * The path to store the job definition in.
         * @param path the value to set
         * @return this builder
         **/
        

public Builder path(String path) {
    this.path = path;
    return this;
}
            /**
     * List of tasks in a job.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("tasks")
private java.util.List<Task> tasks;

        /**
         * List of tasks in a job.
         * @param tasks the value to set
         * @return this builder
         **/
        

public Builder tasks(java.util.List<Task> tasks) {
    this.tasks = tasks;
    return this;
}
            /**
     * An optional list of parameters.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parameters")
private java.util.List<Parameter> parameters;

        /**
         * An optional list of parameters.
         * @param parameters the value to set
         * @return this builder
         **/
        

public Builder parameters(java.util.List<Parameter> parameters) {
    this.parameters = parameters;
    return this;
}
            /**
     * An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeoutSeconds")
private Integer timeoutSeconds;

        /**
         * An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.
         * @param timeoutSeconds the value to set
         * @return this builder
         **/
        

public Builder timeoutSeconds(Integer timeoutSeconds) {
    this.timeoutSeconds = timeoutSeconds;
    return this;
}


        public UpdateJobDetails build() {
            UpdateJobDetails model = new UpdateJobDetails(this.runAs
                , this.name
                , this.description
                , this.schedule
                , this.continuous
                , this.maxConcurrentRuns
                , this.gitConfig
                , this.queue
                , this.jobClusters
                , this.path
                , this.tasks
                , this.parameters
                , this.timeoutSeconds);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UpdateJobDetails model) {
                this.runAs(model.getRunAs());
    this.name(model.getName());
    this.description(model.getDescription());
    this.schedule(model.getSchedule());
    this.continuous(model.getContinuous());
    this.maxConcurrentRuns(model.getMaxConcurrentRuns());
    this.gitConfig(model.getGitConfig());
    this.queue(model.getQueue());
    this.jobClusters(model.getJobClusters());
    this.path(model.getPath());
    this.tasks(model.getTasks());
    this.parameters(model.getParameters());
    this.timeoutSeconds(model.getTimeoutSeconds());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The id with which the job run as.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("runAs")
    private final String runAs;

        /**
     * The id with which the job run as.
     * @return the value
     **/
    
    public String getRunAs() {
        return runAs;
    }


        /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * A description for the job.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * A description for the job.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("schedule")
    private final Schedule schedule;

    
    public Schedule getSchedule() {
        return schedule;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("continuous")
    private final Continuous continuous;

    
    public Continuous getContinuous() {
        return continuous;
    }


        /**
     * Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("maxConcurrentRuns")
    private final Integer maxConcurrentRuns;

        /**
     * Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.
     * @return the value
     **/
    
    public Integer getMaxConcurrentRuns() {
        return maxConcurrentRuns;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("gitConfig")
    private final GitConfig gitConfig;

    
    public GitConfig getGitConfig() {
        return gitConfig;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("queue")
    private final Queue queue;

    
    public Queue getQueue() {
        return queue;
    }


        /**
     * List of job cluster configurations.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("jobClusters")
    private final java.util.List<JobCluster> jobClusters;

        /**
     * List of job cluster configurations.
     * @return the value
     **/
    
    public java.util.List<JobCluster> getJobClusters() {
        return jobClusters;
    }


        /**
     * The path to store the job definition in.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * The path to store the job definition in.
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * List of tasks in a job.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tasks")
    private final java.util.List<Task> tasks;

        /**
     * List of tasks in a job.
     * @return the value
     **/
    
    public java.util.List<Task> getTasks() {
        return tasks;
    }


        /**
     * An optional list of parameters.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parameters")
    private final java.util.List<Parameter> parameters;

        /**
     * An optional list of parameters.
     * @return the value
     **/
    
    public java.util.List<Parameter> getParameters() {
        return parameters;
    }


        /**
     * An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeoutSeconds")
    private final Integer timeoutSeconds;

        /**
     * An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.
     * @return the value
     **/
    
    public Integer getTimeoutSeconds() {
        return timeoutSeconds;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UpdateJobDetails(");
        sb.append("runAs=").append(String.valueOf(this.runAs));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", schedule=").append(String.valueOf(this.schedule));
        sb.append(", continuous=").append(String.valueOf(this.continuous));
        sb.append(", maxConcurrentRuns=").append(String.valueOf(this.maxConcurrentRuns));
        sb.append(", gitConfig=").append(String.valueOf(this.gitConfig));
        sb.append(", queue=").append(String.valueOf(this.queue));
        sb.append(", jobClusters=").append(String.valueOf(this.jobClusters));
        sb.append(", path=").append(String.valueOf(this.path));
        sb.append(", tasks=").append(String.valueOf(this.tasks));
        sb.append(", parameters=").append(String.valueOf(this.parameters));
        sb.append(", timeoutSeconds=").append(String.valueOf(this.timeoutSeconds));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UpdateJobDetails)) {
            return false;
        }

        UpdateJobDetails other = (UpdateJobDetails) o;
        return java.util.Objects.equals(this.runAs, other.runAs) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.schedule, other.schedule) &&
            java.util.Objects.equals(this.continuous, other.continuous) &&
            java.util.Objects.equals(this.maxConcurrentRuns, other.maxConcurrentRuns) &&
            java.util.Objects.equals(this.gitConfig, other.gitConfig) &&
            java.util.Objects.equals(this.queue, other.queue) &&
            java.util.Objects.equals(this.jobClusters, other.jobClusters) &&
            java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.tasks, other.tasks) &&
            java.util.Objects.equals(this.parameters, other.parameters) &&
            java.util.Objects.equals(this.timeoutSeconds, other.timeoutSeconds);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.runAs == null ? 43 : this.runAs.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.schedule == null ? 43 : this.schedule.hashCode());
        result = (result * PRIME) + (this.continuous == null ? 43 : this.continuous.hashCode());
        result = (result * PRIME) + (this.maxConcurrentRuns == null ? 43 : this.maxConcurrentRuns.hashCode());
        result = (result * PRIME) + (this.gitConfig == null ? 43 : this.gitConfig.hashCode());
        result = (result * PRIME) + (this.queue == null ? 43 : this.queue.hashCode());
        result = (result * PRIME) + (this.jobClusters == null ? 43 : this.jobClusters.hashCode());
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.tasks == null ? 43 : this.tasks.hashCode());
        result = (result * PRIME) + (this.parameters == null ? 43 : this.parameters.hashCode());
        result = (result * PRIME) + (this.timeoutSeconds == null ? 43 : this.timeoutSeconds.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/update_job_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UpdateJobDetails(object):
    """
    The data to update a job.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new UpdateJobDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param run_as:
            The value to assign to the run_as property of this UpdateJobDetails.
        :type run_as: str

        :param name:
            The value to assign to the name property of this UpdateJobDetails.
        :type name: str

        :param description:
            The value to assign to the description property of this UpdateJobDetails.
        :type description: str

        :param schedule:
            The value to assign to the schedule property of this UpdateJobDetails.
        :type schedule: oci.aidataplatform_dp.models.Schedule

        :param continuous:
            The value to assign to the continuous property of this UpdateJobDetails.
        :type continuous: oci.aidataplatform_dp.models.Continuous

        :param max_concurrent_runs:
            The value to assign to the max_concurrent_runs property of this UpdateJobDetails.
        :type max_concurrent_runs: int

        :param git_config:
            The value to assign to the git_config property of this UpdateJobDetails.
        :type git_config: oci.aidataplatform_dp.models.GitConfig

        :param queue:
            The value to assign to the queue property of this UpdateJobDetails.
        :type queue: oci.aidataplatform_dp.models.Queue

        :param job_clusters:
            The value to assign to the job_clusters property of this UpdateJobDetails.
        :type job_clusters: list[oci.aidataplatform_dp.models.JobCluster]

        :param path:
            The value to assign to the path property of this UpdateJobDetails.
        :type path: str

        :param tasks:
            The value to assign to the tasks property of this UpdateJobDetails.
        :type tasks: list[oci.aidataplatform_dp.models.Task]

        :param parameters:
            The value to assign to the parameters property of this UpdateJobDetails.
        :type parameters: list[oci.aidataplatform_dp.models.Parameter]

        :param timeout_seconds:
            The value to assign to the timeout_seconds property of this UpdateJobDetails.
        :type timeout_seconds: int

        """
        self.swagger_types = {
            'run_as': 'str',
            'name': 'str',
            'description': 'str',
            'schedule': 'Schedule',
            'continuous': 'Continuous',
            'max_concurrent_runs': 'int',
            'git_config': 'GitConfig',
            'queue': 'Queue',
            'job_clusters': 'list[JobCluster]',
            'path': 'str',
            'tasks': 'list[Task]',
            'parameters': 'list[Parameter]',
            'timeout_seconds': 'int'
        }

        self.attribute_map = {
            'run_as': 'runAs',
            'name': 'name',
            'description': 'description',
            'schedule': 'schedule',
            'continuous': 'continuous',
            'max_concurrent_runs': 'maxConcurrentRuns',
            'git_config': 'gitConfig',
            'queue': 'queue',
            'job_clusters': 'jobClusters',
            'path': 'path',
            'tasks': 'tasks',
            'parameters': 'parameters',
            'timeout_seconds': 'timeoutSeconds'
        }

        self._run_as = None
        self._name = None
        self._description = None
        self._schedule = None
        self._continuous = None
        self._max_concurrent_runs = None
        self._git_config = None
        self._queue = None
        self._job_clusters = None
        self._path = None
        self._tasks = None
        self._parameters = None
        self._timeout_seconds = None

    @property
    def run_as(self):
        """
        Gets the run_as of this UpdateJobDetails.
        The id with which the job run as.


        :return: The run_as of this UpdateJobDetails.
        :rtype: str
        """
        return self._run_as

    @run_as.setter
    def run_as(self, run_as):
        """
        Sets the run_as of this UpdateJobDetails.
        The id with which the job run as.


        :param run_as: The run_as of this UpdateJobDetails.
        :type: str
        """
        self._run_as = run_as

    @property
    def name(self):
        """
        **[Required]** Gets the name of this UpdateJobDetails.
        A user-friendly name. Does not have to be unique, and is changeable.


        :return: The name of this UpdateJobDetails.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this UpdateJobDetails.
        A user-friendly name. Does not have to be unique, and is changeable.


        :param name: The name of this UpdateJobDetails.
        :type: str
        """
        self._name = name

    @property
    def description(self):
        """
        Gets the description of this UpdateJobDetails.
        A description for the job.


        :return: The description of this UpdateJobDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this UpdateJobDetails.
        A description for the job.


        :param description: The description of this UpdateJobDetails.
        :type: str
        """
        self._description = description

    @property
    def schedule(self):
        """
        Gets the schedule of this UpdateJobDetails.

        :return: The schedule of this UpdateJobDetails.
        :rtype: oci.aidataplatform_dp.models.Schedule
        """
        return self._schedule

    @schedule.setter
    def schedule(self, schedule):
        """
        Sets the schedule of this UpdateJobDetails.

        :param schedule: The schedule of this UpdateJobDetails.
        :type: oci.aidataplatform_dp.models.Schedule
        """
        self._schedule = schedule

    @property
    def continuous(self):
        """
        Gets the continuous of this UpdateJobDetails.

        :return: The continuous of this UpdateJobDetails.
        :rtype: oci.aidataplatform_dp.models.Continuous
        """
        return self._continuous

    @continuous.setter
    def continuous(self, continuous):
        """
        Sets the continuous of this UpdateJobDetails.

        :param continuous: The continuous of this UpdateJobDetails.
        :type: oci.aidataplatform_dp.models.Continuous
        """
        self._continuous = continuous

    @property
    def max_concurrent_runs(self):
        """
        Gets the max_concurrent_runs of this UpdateJobDetails.
        Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.


        :return: The max_concurrent_runs of this UpdateJobDetails.
        :rtype: int
        """
        return self._max_concurrent_runs

    @max_concurrent_runs.setter
    def max_concurrent_runs(self, max_concurrent_runs):
        """
        Sets the max_concurrent_runs of this UpdateJobDetails.
        Indicates the number of executions for the same job that can be run concurrently. The maximum value cannot exceed 1000.


        :param max_concurrent_runs: The max_concurrent_runs of this UpdateJobDetails.
        :type: int
        """
        self._max_concurrent_runs = max_concurrent_runs

    @property
    def git_config(self):
        """
        Gets the git_config of this UpdateJobDetails.

        :return: The git_config of this UpdateJobDetails.
        :rtype: oci.aidataplatform_dp.models.GitConfig
        """
        return self._git_config

    @git_config.setter
    def git_config(self, git_config):
        """
        Sets the git_config of this UpdateJobDetails.

        :param git_config: The git_config of this UpdateJobDetails.
        :type: oci.aidataplatform_dp.models.GitConfig
        """
        self._git_config = git_config

    @property
    def queue(self):
        """
        Gets the queue of this UpdateJobDetails.

        :return: The queue of this UpdateJobDetails.
        :rtype: oci.aidataplatform_dp.models.Queue
        """
        return self._queue

    @queue.setter
    def queue(self, queue):
        """
        Sets the queue of this UpdateJobDetails.

        :param queue: The queue of this UpdateJobDetails.
        :type: oci.aidataplatform_dp.models.Queue
        """
        self._queue = queue

    @property
    def job_clusters(self):
        """
        Gets the job_clusters of this UpdateJobDetails.
        List of job cluster configurations.


        :return: The job_clusters of this UpdateJobDetails.
        :rtype: list[oci.aidataplatform_dp.models.JobCluster]
        """
        return self._job_clusters

    @job_clusters.setter
    def job_clusters(self, job_clusters):
        """
        Sets the job_clusters of this UpdateJobDetails.
        List of job cluster configurations.


        :param job_clusters: The job_clusters of this UpdateJobDetails.
        :type: list[oci.aidataplatform_dp.models.JobCluster]
        """
        self._job_clusters = job_clusters

    @property
    def path(self):
        """
        Gets the path of this UpdateJobDetails.
        The path to store the job definition in.


        :return: The path of this UpdateJobDetails.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this UpdateJobDetails.
        The path to store the job definition in.


        :param path: The path of this UpdateJobDetails.
        :type: str
        """
        self._path = path

    @property
    def tasks(self):
        """
        Gets the tasks of this UpdateJobDetails.
        List of tasks in a job.


        :return: The tasks of this UpdateJobDetails.
        :rtype: list[oci.aidataplatform_dp.models.Task]
        """
        return self._tasks

    @tasks.setter
    def tasks(self, tasks):
        """
        Sets the tasks of this UpdateJobDetails.
        List of tasks in a job.


        :param tasks: The tasks of this UpdateJobDetails.
        :type: list[oci.aidataplatform_dp.models.Task]
        """
        self._tasks = tasks

    @property
    def parameters(self):
        """
        Gets the parameters of this UpdateJobDetails.
        An optional list of parameters.


        :return: The parameters of this UpdateJobDetails.
        :rtype: list[oci.aidataplatform_dp.models.Parameter]
        """
        return self._parameters

    @parameters.setter
    def parameters(self, parameters):
        """
        Sets the parameters of this UpdateJobDetails.
        An optional list of parameters.


        :param parameters: The parameters of this UpdateJobDetails.
        :type: list[oci.aidataplatform_dp.models.Parameter]
        """
        self._parameters = parameters

    @property
    def timeout_seconds(self):
        """
        Gets the timeout_seconds of this UpdateJobDetails.
        An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.


        :return: The timeout_seconds of this UpdateJobDetails.
        :rtype: int
        """
        return self._timeout_seconds

    @timeout_seconds.setter
    def timeout_seconds(self, timeout_seconds):
        """
        Sets the timeout_seconds of this UpdateJobDetails.
        An optional value to indicate the max run duration of a job after which job will be timed out. The default is Zero indicating no timeout value.


        :param timeout_seconds: The timeout_seconds of this UpdateJobDetails.
        :type: int
        """
        self._timeout_seconds = timeout_seconds

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
